syntax = "proto3";

package raft;

service Raft {
  rpc RequestVote (VoteRequest) returns (VoteResponse);
  rpc AppendEntries (AppendEntriesRequest) returns (AppendEntriesResponse);
}

message VoteRequest {
  int32 term = 1;
  string candidateId = 2;
}

message VoteResponse {
  int32 term = 1;
  bool voteGranted = 2;
}

message LogEntry {
  string operation = 1;
  int32 term = 2;
  int32 index = 3;
}

message AppendEntriesRequest {
  int32 term = 1;
  string leaderId = 2;
}

message AppendEntriesResponse {
  int32 term = 1;
  bool success = 2;
}
